src/common/stats/StatsManager.cpp

```diff
@@ -311,6 +323,124 @@ StatusOr<StatsManager::VT> StatsManager::readHisto(const std::string& counterNam
     return sm.histograms_[index].second->getPercentileEstimate(pct, level);
 }
 
+
+// static
+StatusOr<StatsManager::ParsedName>
+StatsManager::parseMetricName(folly::StringPiece metricName) {
+    std::vector<std::string> parts;
+    folly::split(".", metricName, parts, true);
+    if (parts.size() != 3) {
+        LOG(ERROR) << "\"" << metricName << "\" is not a valid metric name";
+        return Status::Error("\"%s\" is not a valid metric name", metricName.data());
+    }
+
+    std::string name = parts[0];
+    TimeRange range;
+    if (parts[2] == "60") {
+        range = TimeRange::ONE_MINUTE;
+    } else if (parts[2] == "600") {
+        range = TimeRange::TEN_MINUTES;
+    } else if (parts[2] == "3600") {
+        range = TimeRange::ONE_HOUR;
+    } else {
+        // Unsupported time range
+        LOG(ERROR) << "Unsupported time range \"" << parts[2] << "\"";
+        return Status::Error(folly::stringPrintf("Unsupported time range \"%s\"",
+                                                 parts[2].c_str()));
+    }
+
+    StatsMethod method;
+    folly::toLowerAscii(parts[1]);
+    if (parts[1] == "sum") {
+        method = StatsMethod::SUM;
+    } else if (parts[1] == "count") {
+        method = StatsMethod::COUNT;
+    } else if (parts[1] == "avg") {
+        method = StatsMethod::AVG;
+    } else if (parts[1] == "rate") {
+        method = StatsMethod::RATE;
+    // } else if (parts[1][0] == 'p') {   // TODO(shylock)
+    } else {
+        LOG(ERROR) << "Unsupported statistic method \"" << parts[1] << "\"";
+        return Status::Error(folly::stringPrintf("Unsupported statistic method \"%s\"",
+                                                 parts[1].c_str()));
+    }
+
+    return StatsManager::ParsedName {
+        name,
+        method,
+        range,
+    };
+}
+
+
+// The raw metric format based on JSON
+// 1. Gauge, time serial value
+// 2. Histogram, time serial value distribution
+// The raw data can be transformed to multiple specified foramt (various user-defined metric format)
+// E.G.
+// {
+//     "Gauges": [...],
+//     "Histograms:" [...],
+// }
+// Gauge
+// {
+//     "name": "xxxxx",
+//     "value": 33,
+//     "labels": ["name:nebula", "type:qps"]  // key:value
+// }
+// Histogram
+// {
+//     "name": "xxxxx",
+//     "value_range": [0, 100],
+//     "buckets": [2, 3, 0, 11, ...],
+//     "labels": ["name:nebula", "type:latency"]
+// }
```

Comment:
Shall we need to format our metrics to these format ? Would it be better to do it in nebula_exporter outside of nebula.